server/src/test/kotlin/org/itmodreamteam/myrest/server/service/restaurant/ReservationTest.kt

```diff
@@ -0,0 +1,148 @@
+        user2 = userRepository.save(User("Sam", "Ivanov"))
+        user3 = userRepository.save(User("Karl", "Marks"))
+        manager = employeeRepository.save(Manager(restaurant, user3))
+        val now = LocalDateTime.now()
+        authorize(user1)
+        reservation = reservationRepository.save(Reservation(user1, table, now.plusHours(1), now.plusHours(5)))
+    }
+
+    private fun authorize(user: User) {
+        Mockito.`when`(currentUserService.currentUserEntity).thenReturn(user)
+    }
+
+    @Test
+    fun `When create new reservation, then new reservation created`() {
+        val reservation = reservationRepository.findByIdOrNull(reservation.id)
+        Assertions.assertThat(reservation).isNotNull
+        Assertions.assertThat(reservation?.status).isEqualTo(ReservationStatus.PENDING)
+        Assertions.assertThat(reservation?.user).isEqualTo(user1)
+        Assertions.assertThat(reservation?.table).isEqualTo(table)
+    }
+
+    @Test
+    fun `When create new reservation right after existing one, then new a reservation created`() {
+        authorize(user2)
+        val reservation = reservationService.submitReservationForApproval(
+            table,
+            reservation.activeUntil,
+            reservation.activeUntil.plusHours(2)
+        )
+        Assertions.assertThat(reservation).isNotNull
+        Assertions.assertThat(reservation.status).isEqualTo(ReservationStatus.PENDING)
+        Assertions.assertThat(reservation.user).isEqualTo(user2)
+        Assertions.assertThat(reservation.table).isEqualTo(table)
+    }
+
+    @Test(expected = UserException::class)
+    fun `Given existing reservation, when create reservation to same table and overlapping time, then failure`() {
+        authorize(user2)
+        reservationService.submitReservationForApproval(
+            table,
+            reservation.activeFrom.minusHours(2),
+            reservation.activeFrom.plusMinutes(1)
+        )
+    }
+
+    @Test
+    fun `Given rejected existing reservation, when create reservation to same table and time, then a reservation created`() {
+        authorize(user1)
+        reservationService.reject(reservation)
+        reservationService.submitReservationForApproval(table, reservation.activeFrom, reservation.activeUntil)
+    }
+
+    @Test(expected = UserException::class)
+    fun `When try to reject other user's reservation, then failure`() {
+        authorize(user2)
+        reservationService.reject(reservation)
+    }
+
+    @Test
+    fun `Given existing reservation in state PENDING, when manager approves it, then reservation is APPROVED`() {
+        authorize(user3)
+        val approved = reservationService.approve(reservation)
+        Assertions.assertThat(approved.status).isEqualTo(ReservationStatus.APPROVED)
+    }
+
+    @Test(expected = UserException::class)
+    fun `Given existing PENDING reservation and manager who is not responsible to the reservation, when try to approve it, then failure`() {
```

Comment:
responsible for

Reply:
As I see, user1 is not manager at all. So, should it sound like: "and user who is not manager" ?
Then, we may need one more test for approving reservation by manager from a different restaurant.